### ISSUE TITLE
[Bug] UI display priority

### ISSUE THREAD

**Issue by MagicHatJo** [comfyanonymous/ComfyUI#1328]
### Problem:
Currently, z-index of <textarea> increments linearly, which places it over other boxes and eventually, even the menu.

Examples:
![chrome_BxMTQCsQx2](https://github.com/comfyanonymous/ComfyUI/assets/38716012/be0caa13-706b-4e04-bf85-84c07b8e8d2e)
![chrome_o0n9swzttq](https://github.com/comfyanonymous/ComfyUI/assets/38716012/49b688de-3c46-4afa-8537-5a399a85cf84)
![chrome_C874GtlsRS](https://github.com/comfyanonymous/ComfyUI/assets/38716012/4233dd12-171a-4d60-b92b-cdf18b6ce1d3)
![chrome_JVEW5OJXqC](https://github.com/comfyanonymous/ComfyUI/assets/38716012/a23a3c19-1bd2-4541-80de-28169c0fb2f3)
![chrome_vjgCFImDQb](https://github.com/comfyanonymous/ComfyUI/assets/38716012/e84eb4c2-d049-459a-b5a4-b06b8055576b)

### Expected:
Properties, pop ups, and menu should be above node and node contents.
![chrome_dWrvmagEMT](https://github.com/comfyanonymous/ComfyUI/assets/38716012/3517dc46-b1a3-45fe-b6aa-c2669019989e)

----------------------------------------------------------------------------------
For now, I've set custom user css locally to specifically move menu and properties window to 999 and 1000, but this doesn't seem like a good modular solution. Is there a reason for z-index to continually increment, or is this a bug? I can work on this issue if I get a conclusive answer, I'm just not sure if I'm overlooking something that necessitates said incrementations.

**Comment by wallish77** [comment]
Can confirm that I have seen this issue.  Often happens when attempting to rename a node (node rename box ends up behind the node)

**Comment by szwang-seedtech** [comment]
Is there any solution? I believe it only happens on the text widget

**Comment by ltdrdata** [comment]
> Is there any solution? I believe it only happens on the text widget

There is no solution yet.
This is a problem that many developers are aware of, already. However, it's an issue that they find difficult to tackle readily because it requires modifications at a very foundational level of the code.

**Comment by szwang-seedtech** [comment]
Or maybe we can set the nodes never to be overlapped, something like collision volume? How about that?

**Comment by dchatel** [comment]
The culprit may be related to line 267 of file web/scripts/domWidgets.js:
```javascript
zIndex: app.graph._nodes.indexOf(node),
```
when I commented this line, it kinda helped a little bit, but this is not the only problem.
Another issue I've found is that when you collapse a node which contains a textarea, its css property display very quickly gets to "none" then null, instead of staying to "none". I haven't found where exactly that happens in the code, but if you insist with the browser dev tools, you can see the behavior.

**Comment by Trung0246** [comment]
Yeah I think this is basically unfixable since the text box is an html DOM object which to be rendered by the browser on top of the canvas graph object (which is another separate DOM object).

 Therefore I think the actual fix would be just create a custom widget that purely use canvas API instead of relying on any other DOM objects.

**Comment by dchatel** [comment]
Recently, this bug has gotten worse.
https://github.com/comfyanonymous/ComfyUI/assets/1372055/0c9a2c2a-6f9e-412d-9b90-cfcb724e4222
